imports/api/users.js

```diff
@@ -192,13 +207,27 @@ Meteor.methods({
    * update profile image with new image in S3 collection
    * @param {String} profileImageUrl
    */
-  'users.updateProfileImage' (profileImageUrl) {
+  'users.updateProfileImage' (profileImageUrl, storageType) {
     check(profileImageUrl, String)
+    if (storageType === 'AWS') {
+      if (profileImageUrl.endsWith('/thumbnail')) profileImageUrl = profileImageUrl.slice(0, -10) + '/image'
+      else if (!profileImageUrl.endsWith('/image')) profileImageUrl = profileImageUrl + '/image' 
+    } 
     return Meteor.users.update({ _id: Meteor.userId() }, {
       '$set': { 'profile.profileImage': profileImageUrl }
     })
   },
 
+  'users.updateProfileThumbnail' (profileImageUrl, storageType) {
```

Comment:
same thing, check that it's a string, and set it directly. Handle the update in the getURL